src/net/fe/builderStage/ShopInventory.java

```diff
@@ -0,0 +1,118 @@
+package net.fe.builderStage;
+import java.util.Dictionary;
+import java.util.HashMap;
+import java.util.Map;
+
+import net.fe.builderStage.ShopInventory;
+import net.fe.modifier.Modifier;
+import net.fe.unit.Item;
+import net.fe.unit.Unit;
+import net.fe.unit.Weapon;
+
+public class ShopInventory {
+	private static ShopInventory obj;
+
+	private Map<String, Integer> inventory;
+
+	private ShopInventory(Iterable<Modifier> mods){
+		inventory = new HashMap<String, Integer>();
+
+		for(Item i : Item.getAllItems()){
+			inventory.put(i.name, -1);
+		}
+
+		if(mods != null){
+
+			for(Modifier m : mods){
+				m.setShopLimits(inventory);
+			}
+		}
+	}
+
+	/**
+	 * Singleton constructor method
+	 * 
+	 * @param mods The modifiers used by this session.  Only loaded the first time this is called, but
+	 * there's no harm in passing it again.
+	 * 
+	 * @return
+	 */
+	public static ShopInventory GetInstance(Iterable<Modifier> mods){
+		if(obj == null){
+			obj = new ShopInventory(mods);
+		}
+
+		return obj;
+	}
+
+	/**
+	 * destroys the current instance of the inventory.  To be called when the battle starts to reset
+	 * it for the next game.
+	 */
+	public void DestroyInstance(){
```

Comment:
A new TeamBuilderStage is initialized once per team-creation-phase, the shop inventory becomes relevant precisely when the TeamBuilderStage is created, and the shop inventory ceases to be relevant exactly when the TeamBuilderStage ceases to be relevant. And, besides it makes sense that the TeamBuilder would keep track of all limitations related to building a team, including the shop inventory.

So, I'd make an instance of ShopInventory a member of TeamBuilderStage, that instance being created in TeamBuilderStage's constructor and passed down through each new UnitBuilderStage to its ShopMenu whenever TeamBuilderStage creates a new UnitBuilderStage

-------

And then I look closer, and a ShopInventory *is* a member of TeamBuilderStage, and yet UnitBuilderStage/ShopMenu still asks a global for a ShopInventory, instead of asking TeamBuilderStage. How did that come to be?

There is absolutely no requirements for interface compatibility here. ^(Well, the map and team files should remain loadable, but that's disjoint to this.) You are allowed to add or change method parameters.